### PR TITLE
use akka.persistence.cassandra as the config namespace

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Follow up on #81

**Comment by patriknw** [comment]
Thanks for the suggestion @ennru

File: core/src/main/resources/reference.conf [hunk 1]
```diff
@@ -466,23 +466,25 @@ cassandra-plugin {
...
   }
 }
 
+
+
 //#profile
 datastax-java-driver {
   profiles {
-    cassandra-plugin {
+    akka-persistence-cassandra {
```
Comment (ennru) [inline core/src/main/resources/reference.conf]: This difference alone makes this change worth it.
I wonder if adding `-profile` here would make it even more clear.

Comment (patriknw) [reply core/src/main/resources/reference.conf]: I don't mind but it's somewhat already in scope via the `profiles {` and used in `read-profile =`

Comment (ennru) [reply core/src/main/resources/reference.conf]: Here it is clear, but not immediately obvious from where it is referenced in `akka.persistence.cassandra` scope. The `.` vs `-` difference is quite subtile.

Comment (patriknw) [reply core/src/main/resources/reference.conf]: ok, changed

File: core/src/test/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotCleanupSpec.scala [hunk 2]
```diff
@@ -18,7 +18,7 @@ import akka.event.Logging
 
 class CassandraSnapshotCleanupSpec extends CassandraSpec {
 
-  private val configPath = "cassandra-plugin"
+  private val configPath = "akka.persistence.cassandra"
```
Comment (ennru) [inline core/src/test/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotCleanupSpec.scala]: BTW In Alpakka we'd add a `ConfigPath` constant to the `PluginSettings` object.

File: core/src/main/resources/reference.conf [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
-# setting to "cassandra-plugin.journal".
+# setting to "akka.persistence.cassandra.journal".
 #
 # Otherwise you need to create differently named sections containing
 # only those settings that shall be different from the defaults
 # configured here, importing the defaults like so:
 #
-#   another-cassandra-plugin = ${cassandra-plugin}
+#   another-cassandra-plugin = ${akka.persistence.cassandra}
 #   another-cassandra-plugin {
```
[the remark this answers is not in the document]
Comment (patriknw) [reply core/src/main/resources/reference.conf]: that's quite magical, I didn't even know it was possible. I think I prefer the two steps

File: core/src/main/scala/akka/persistence/cassandra/EventsByTagMigration.scala [hunk 4]
```diff
@@ -95,13 +95,13 @@ object EventsByTagMigration {
...
- * @param journalNamespace The config namespace where the journal is configured, default is `cassandra-plugin`
+ * @param journalNamespace The config namespace where the journal is configured, default is `akka.persistence.cassandra`
  * @param readJournalNamespace The config namespace where the query-journal is configured, default is the same
  *                             as `CassandraReadJournal.Identifier`
  */
 class EventsByTagMigration(
     system: ActorSystem,
-    journalNamespace: String = "cassandra-plugin",
+    journalNamespace: String = "akka.persistence.cassandra",
     readJournalNamespace: String = CassandraReadJournal.Identifier)
```
[the remark this answers is not in the document]
Comment (patriknw) [reply core/src/main/scala/akka/persistence/cassandra/EventsByTagMigration.scala]: it's correct but I simplified it now to one single parameter `pluginConfigPath: String = "akka.persistence.cassandra"`

File: core/src/test/scala/akka/persistence/cassandra/CassandraSpec.scala [hunk 5]
```diff
@@ -247,7 +247,7 @@ abstract class CassandraSpec(
         Long.MaxValue,
         100,
         None,
-        readProfile = "cassandra-plugin",
+        readProfile = "akka.persistence.cassandra",
```
[the remark this answers is not in the document]
Comment (patriknw) [reply core/src/test/scala/akka/persistence/cassandra/CassandraSpec.scala]: thx

File: core/src/test/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotCleanupSpec.scala (hunk 2 above)
Comment (patriknw) [reply core/src/test/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotCleanupSpec.scala]: I did some additional cleanup related to that

File: docs/src/main/paradox/journal.md [hunk 6]
```diff
@@ -62,7 +62,7 @@ Any other consistency level is highly discouraged.
 
 ```
 datastax-java-driver.profiles {
-  cassandra-plugin {
+  akka.persistence.cassandra {
```
[the remark this answers is not in the document]
Comment (patriknw) [reply docs/src/main/paradox/journal.md]: yes. thanks

**Pull request merged**